web/tests/conftest.py

```
@@ -497,17 +529,32 @@ def _set_valid_licence(app):
 
 
 def _add_valid_checklist(app):
-    app.checklist = SILChecklist.objects.create(
-        import_application=app,
-        case_update=YesNoNAChoices.yes,
-        fir_required=YesNoNAChoices.yes,
-        response_preparation=True,
-        validity_period_correct=YesNoNAChoices.yes,
-        endorsements_listed=YesNoNAChoices.yes,
-        authorisation=True,
-        authority_required=YesNoNAChoices.yes,
-        authority_received=YesNoNAChoices.yes,
-        authority_cover_items_listed=YesNoNAChoices.yes,
-        quantities_within_authority_restrictions=YesNoNAChoices.yes,
-        authority_police=YesNoNAChoices.yes,
-    )
+    checklist = {
+        "import_application": app,
+        "case_update": YesNoNAChoices.yes,
+        "fir_required": YesNoNAChoices.yes,
+        "response_preparation": True,
+        "validity_period_correct": YesNoNAChoices.yes,
+        "endorsements_listed": YesNoNAChoices.yes,
+        "authorisation": True,
+    }
+    match app.process_type:
+        case ProcessTypes.FA_SIL:
+            app.checklist = SILChecklist.objects.create(
+                **checklist
+                | {
+                    "authority_required": YesNoNAChoices.yes,
+                    "authority_received": YesNoNAChoices.yes,
+                    "authority_cover_items_listed": YesNoNAChoices.yes,
+                    "quantities_within_authority_restrictions": YesNoNAChoices.yes,
+                    "authority_police": YesNoNAChoices.yes,
+                }
+            )
+        case ProcessTypes.FA_DFL:
+            app.checklist = DFLChecklist.objects.create(
+                **checklist
+                | {
+                    "deactivation_certificate_attached": YesNoNAChoices.yes,
+                    "deactivation_certificate_issued": YesNoNAChoices.yes,
+                }
+            )
```

If we are intentionally not creating checklists for any other app type lets raise a value error.
e.g.
```python
case _:
    raise ValueError(f"Invalid process_type: {app.process_type}")
```